File: CleanJava/app/src/debug/java/com/vascome/fogtail/screens/dev_settings/views/DeveloperSettingsView.java
package com.vascome.fogtail.screens.dev_settings.views;

import android.support.annotation.AnyThread;
import android.support.annotation.NonNull;

import okhttp3.logging.HttpLoggingInterceptor;

/**
 * Created by vasilypopov on 11/23/17
 * Copyright (c) 2017 MVPJava. All rights reserved.
 */

public interface DeveloperSettingsView {

    @AnyThread
    void changeGitSha(@NonNull String gitSha);

    @AnyThread
    void changeBuildDate(@NonNull String date);

    @AnyThread
    void changeBuildVersionCode(@NonNull String versionCode);

    @AnyThread
    void changeBuildVersionName(@NonNull String versionName);

    @AnyThread
    void changeStethoState(boolean enabled);

    @AnyThread
    void changeLeakCanaryState(boolean enabled);

    @AnyThread
    void changeTinyDancerState(boolean enabled);

    @AnyThread
    void changeHttpLoggingLevel(@NonNull HttpLoggingInterceptor.Level loggingLevel);

    @AnyThread
    void showMessage(@NonNull String message);

    @AnyThread
    void showAppNeedsToBeRestarted();

}

File: MVVMKotlin/app/src/debug/java/com/vascome/fogtail/presentation/dev_settings/views/DeveloperSettingsView.kt
package com.vascome.fogtail.presentation.dev_settings.views

import android.support.annotation.AnyThread

import okhttp3.logging.HttpLoggingInterceptor

/**
 * Created by vasilypopov on 11/23/17
 * Copyright (c) 2017 MVPJava. All rights reserved.
 */

interface DeveloperSettingsView {

    @AnyThread
    fun changeGitSha(gitSha: String)

    @AnyThread
    fun changeBuildDate(date: String)

    @AnyThread
    fun changeBuildVersionCode(versionCode: String)

    @AnyThread
    fun changeBuildVersionName(versionName: String)

    @AnyThread
    fun changeStethoState(enabled: Boolean)

    @AnyThread
    fun changeLeakCanaryState(enabled: Boolean)

    @AnyThread
    fun changeTinyDancerState(enabled: Boolean)

    @AnyThread
    fun changeHttpLoggingLevel(loggingLevel: HttpLoggingInterceptor.Level)

    @AnyThread
    fun showMessage(message: String)

    @AnyThread
    fun showAppNeedsToBeRestarted()

}
